CONTRIBUTING.md

```diff
@@ -1,61 +1,520 @@
 # Contributing Guidelines
```

Comment:
One thought I had was keeping the various codebases we have sync'd as teams iterate on their guides. Should we consider extracting common elements (e.g., the design philosophy, code of conduct, how to file bugs) in some central location on the Amplify site and focus these guides on the mechanics of each codebase?

Reply:
Hm. Yea. This is worth exploring. It's a cost-benefit sort of question. If the sections are smaller than \<meta-metric _m_\>, it probably makes sense to just keep a few flat forks. But if larger than \<meta-metric _m_\>, it may make sense to add the tooling, additional repos, and process complexity to include some stubs from a central location.

The metric might be lines of text, or amount of time we're spending on the docs. Or, the amount of fear we have that they'll drift, and our perception of how bad that would be for the project?

Reply:
(See discussion below @undefobj & @jamesonwilliams discuss doing this section in the docs project, instead of here, and then just linking out to it.)